lib/ansible/module_utils/eos.py

```diff
@@ -180,6 +186,17 @@ def _load_params(self):
             if key in NET_COMMON_ARGS.keys():
                 if not params.get(key) and value is not None:
                     params[key] = value
+
+        env_overrides = dict(
+            username='ANSIBLE_NET_USERNAME',
+            password='ANSIBLE_NET_PASSWORD',
+            ssh_keyfile='ANSIBLE_NET_SSH_KEYFILE',
+            authorize='ANSIBLE_NET_AUTHORIZE',
+            auth_pass='ANSIBLE_NET_AUTH_PASS',
+        )
+        for key, env_var in env_overrides.items():
+            if env_var in os.environ:
+                params[key] = os.environ[env_var]
```

Comment:
I thought this way too at first, and that's how my very first attempt went, but upon rereading @privateip's request, I think this is the intended order. Specifically, the precedence order given to me was:

1. Env
2. Task arg
3. Provider arg

This code should apply the values in that order; the task arguments are initially in place, then the provider arguments fill in any gaps, and finally the environment variables clobber anything if a value is set. This is also specifically why I don't use `environ.get()` here, I don't want to replace existing parameters with `None`.